cmd/compareBedDistanceBasedOnName/compareBedDistanceBasedOnName.go

```diff
@@ -0,0 +1,102 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/vertgenlab/gonomics/bed"
+	"github.com/vertgenlab/gonomics/fileio"
+	"log"
+)
+
+func compareBedDistanceBasedOnName(inputBed string, genomeBed string, outBed string) {
+	out := fileio.EasyCreate(outBed)
+	input := bed.Read(inputBed)
+	genome := bed.Read(genomeBed)
+
+	genomeMap := make(map[string][]bed.Bed)
+	var i int
+
+	for i = range genome {
+		genomeMap[genome[i].Name] = append(genomeMap[genome[i].Name], genome[i])
+	}
+
+	var inputNameField []bed.Bed
+	var found bool
+	var distance int
+	var lowestDistance int
+	var err error
+	var j int
+
+	for i := 0; i < len(input); i++ {
+		lowestDistance = 0
+		inputNameField, found = genomeMap[input[i].Name]
+		if found != true {
+			log.Fatalf("Did not find %s", input[i].Name)
+		}
+		for j = range inputNameField {
+			distance, err = bed.CompareDistance(inputNameField[j], input[i])
```

Comment:
do we need to catch this err in the loop at all? I noticed it is checked once we exit the loop, but not for each iteration